generic_server.cc

```diff
@@ -285,23 +288,48 @@ future<> server::shutdown() {
     if (_gate.is_closed()) {
         co_return;
     }
-    _all_connections_stopped = _gate.close();
+
+    shared_future<> connections_stopped{_gate.close()};
+     _all_connections_stopped = connections_stopped.get_future();
+
+     // Stop all listeners.
     size_t nr = 0;
     size_t nr_total = _listeners.size();
     _logger.debug("abort accept nr_total={}", nr_total);
     for (auto&& l : _listeners) {
         l.abort_accept();
         _logger.debug("abort accept {} out of {} done", ++nr, nr_total);
     }
+
+    // Shutdown RX side of the connections, so no new requests could be received.
+    // Leave the TX side so the responses to ongoing requests could be sent.
+    _logger.debug("Shutting down RX side of {} connections", _connections_list.size());
+    for (auto& connection : _connections_list) {
+        if (!connection.shutdown_input()) {
+            // If failed to shutdown the input side, then attempt to shutdown the output side which should do a complete shutdown of the connection.
+            connection.shutdown_output();
+        }
```

Comment:
This list can have tens of thousands of connections. We may need to yield here.

This is complicated since we may need to prevent races with connections leaving on their own or being added.

Reply:
> Ok, will add yields. just to mention the previous implementation did take that in to account either. 

Then it's okay to defer to a later patch. 

> At this point we have shut down TCP listener sockets, so I believe new connections cannot be added, we only need to take care of the ones that could be deleted while we yielding.

We initiated the shutdown, but until we collect the result futures, it's not done.